src/common/config_obs.h

```diff
@@ -31,17 +32,26 @@ template<class ConfigProxy>
 class md_config_obs_impl {
 public:
   virtual ~md_config_obs_impl() {}
+
   /** @brief Get a table of strings specifying the configuration keys in which the object is interested.
    * This is called when the object is subscribed to configuration changes with add_observer().
    * The returned table should not be freed until the observer is removed with remove_observer().
    * Note that it is not possible to change the set of tracked keys without re-subscribing. */
-  virtual const char** get_tracked_conf_keys() const = 0;
+  virtual const char** get_tracked_conf_keys() const {
+    return nullptr;
+  }
+
+  /**
+   * a leaner version of get_tracked_conf_keys() that returns a single key.
+   * If returns a valid string - get_tracked_conf_keys() will not be called.
+   */
+  virtual std::optional<std::string> get_tracked_conf_key() const noexcept {
```

Comment:
Is there a plan to deprecate / drop the old `get_tracked_conf_keys()`?
How much of work this would need?

Reply:
Not really, as most of the users register multiple keys.
I do want to change the signatures to 'const char * const *' - to better handle const strings as arguments.
But it might be a good idea to modify the multi-key interface altogether - to use strings and be copied directly into the strings of the config manager.

Reply:
But that's for the next PR...

Reply:
I'm not really seeing a big benefit for this PR while it complicates the interface. It's not like we have a ton of config obs which only care about one option name?

I do agree something like `std::array<std::string> const&` would be preferable as a return type.

Reply:
@batrick 
> we have a ton of config obs which only care about one option name?

Yes - we will have (in the OSD), as we plane to move more 'change aware' configs to be wrapped in the single-key md_config_cacher_t (currently we have 11 of those, but that's just the beginning). The reason for this direction is that we've learned that the get<> interface for the configs is very slow.

> std::array<std::string> const&

(I do not think we can use std::array, but std::vector is a valid choice). 
I did not want to have to change all clients of the existing interface in this PR, and preferred to make that the next step. Not sure about the 'const&', though. I will have to play with that. If possible, I'd wish the actual texts to be moved-to during registration.

Reply:
I have modified the PR, discarding the one-key API. See comment below.